dpgen2/op/run_nvnmd_train.py

```
@@ -0,0 +1,670 @@
+        )
+        train_qnn_dict = RunNvNMDTrain.write_other_to_input_script(
+            train_dict,
+            config,
+            do_init_model,
+            True,
+            major_version,
+        )
+
+        with set_directory(work_dir):
+            # open log
+            fplog = open("train.log", "w")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)